library/src/main/java/com/phillipcalvin/iconbutton/IconButton.java

```diff
@@ -65,34 +66,47 @@ protected void onLayout(boolean changed, int left, int top, int right, int botto
         textPaint.getTextBounds(text, 0, text.length(), bounds);
 
         int textWidth = bounds.width();
-        int contentWidth = drawableWidth + iconPadding + textWidth;
-
+        int factor = (drawablePosition == DrawablePositions.BOTH) ? 2 : 1;
+        int contentWidth = drawableWidth + iconPadding * factor + textWidth;
         int contentLeft = (int) ((getWidth() / 2.0) - (contentWidth / 2.0));
+
         setCompoundDrawablePadding(-contentLeft + iconPadding);
+
         switch (drawablePosition) {
             case LEFT:
-                setPadding(contentLeft, 0, 0, 0);
+                setPadding(contentLeft, getPaddingTop(), 0, getPaddingBottom());
                 break;
+
             case RIGHT:
-                setPadding(0, 0, contentLeft, 0);
+                setPadding(0, getPaddingTop(), contentLeft, getPaddingBottom());
                 break;
+
+            case BOTH:
+                setPadding(contentLeft, getPaddingTop(), contentLeft, getPaddingBottom());
+                break;
+
             default:
-                setPadding(0, 0, 0, 0);
+                setPadding(0, getPaddingTop(), 0, getPaddingBottom());
         }
     }
 
     @Override
     public void setCompoundDrawablesWithIntrinsicBounds(Drawable left, Drawable top, Drawable right, Drawable bottom) {
         super.setCompoundDrawablesWithIntrinsicBounds(left, top, right, bottom);
-        if (null != left) {
+
+        if (left != null && right != null) {
+            drawableWidth = left.getIntrinsicWidth() + right.getIntrinsicWidth();
+            drawablePosition = DrawablePositions.BOTH;
+        }else if (left != null) {
```

Comment:
hater!

jk :). i'm so used to running my reformatter on the whole file. i couldn't on this file cause it would rearrange the ordering of methods, so this sucker got left out.